src/main/java/com/hazelcast/kubernetes/KubernetesApiEndpointResolver.java

```diff
@@ -56,49 +53,59 @@
+            return getSimpleDiscoveryNodes(client.endpointsByName(serviceName));
         } else if (serviceLabel != null && !serviceLabel.isEmpty()) {
-            return getSimpleDiscoveryNodes(client.endpointsByLabel(namespace, serviceLabel, serviceLabelValue));
+            return getSimpleDiscoveryNodes(client.endpointsByLabel(serviceLabel, serviceLabelValue));
         }
-        return getSimpleDiscoveryNodes(client.endpoints(namespace));
+        return getSimpleDiscoveryNodes(client.endpoints());
     }
 
-    private List<DiscoveryNode> getSimpleDiscoveryNodes(Endpoints endpoints) {
+    private List<DiscoveryNode> getSimpleDiscoveryNodes(List<Endpoint> endpoints) {
         List<DiscoveryNode> discoveredNodes = new ArrayList<DiscoveryNode>();
-        resolveNotReadyAddresses(discoveredNodes, endpoints.getNotReadyAddresses());
-        resolveAddresses(discoveredNodes, endpoints.getAddresses());
+        resolveAddresses(discoveredNodes, endpoints);
         return discoveredNodes;
     }
 
-    private void resolveNotReadyAddresses(List<DiscoveryNode> discoveredNodes, List<EntrypointAddress> notReadyAddresses) {
+    private void resolveNotReadyAddresses(List<DiscoveryNode> discoveredNodes, List<Endpoint> notReadyAddresses) {
         if (Boolean.TRUE.equals(resolveNotReadyAddresses)) {
             resolveAddresses(discoveredNodes, notReadyAddresses);
         }
     }
 
-    private void resolveAddresses(List<DiscoveryNode> discoveredNodes, List<EntrypointAddress> addresses) {
-        for (EntrypointAddress address : addresses) {
+    private void resolveAddresses(List<DiscoveryNode> discoveredNodes, List<Endpoint> addresses) {
+        for (Endpoint address : addresses) {
             addAddress(discoveredNodes, address);
         }
     }
 
-    private void addAddress(List<DiscoveryNode> discoveredNodes, EntrypointAddress entrypointAddress) {
-        String ip = entrypointAddress.getIp();
-        InetAddress inetAddress = mapAddress(ip);
-        int port = port(entrypointAddress);
-        Address address = new Address(inetAddress, port);
-        discoveredNodes.add(new SimpleDiscoveryNode(address, entrypointAddress.getAdditionalProperties()));
-        if (logger.isFinestEnabled()) {
-            logger.finest("Found node service with address: " + address);
+    private void addAddress(List<DiscoveryNode> discoveredNodes, Endpoint endpoint) {
+        if (Boolean.TRUE.equals(resolveNotReadyAddresses) || endpoint.isReady()) {
+            Address address = createAddress(endpoint.getPrivateAddress());
+            Address publicAddress = createAddress(endpoint.getPublicAddress());
+            discoveredNodes
+                    .add(new SimpleDiscoveryNode(address, publicAddress, endpoint.getAdditionalProperties()));
+            if (logger.isFinestEnabled()) {
+                logger.finest("Found node service with address: " + address);
```

Comment:
why do we only log privateAddress now? log statement can be updated.

Reply:
Updated